examples/test_odx/test_server_odx.py

```diff
@@ -0,0 +1,73 @@
+import logging
+import time
+import typing
+import unittest
+from copy import copy
+
+from uds import IsoServices
+
+import pykiso
+from pykiso.auxiliaries import uds_server_aux
+
+# helper objects to build callbacks can be imported from the pykiso lib
+from pykiso.lib.auxiliaries.udsaux import UdsCallback, UdsServerAuxiliary
+from pykiso.lib.auxiliaries.udsaux.common.uds_response import (
+    NegativeResponseCode,
+)
+
+UDS_CALLBACKS = [
+    UdsCallback(request=0x1003, response=0x5003),
+    # UdsCallback(
+    #     request={
+    #         "service": IsoServices.ReadDataByIdentifier,
+    #         "data": {"parameter": "SoftwareVersion"},
+    #     },
+    #     # also works without response
+    #     response={"SoftwareVersion": "0.17.0"},
+    # ),
+    UdsCallback(
+        request={
+            "service": IsoServices.ReadDataByIdentifier,
+            "data": {"parameter": "SoftwareVersion"},
+        },
+        response={"NEGATIVE": NegativeResponseCode.SERVICE_NOT_SUPPORTED},
+    ),
+]
+
+
+@pykiso.define_test_parameters(suite_id=1, case_id=1, aux_list=[uds_server_aux])
+class ExampleUdsServerTest(pykiso.BasicTest):
+    def setUp(self):
+        """Register various callbacks for the test."""
+        # register all callbacks defined above
+        for callback in UDS_CALLBACKS:
+            uds_server_aux.register_callback(callback)
+        logging.info("-----> Waiting for requests")
+
+    def test_run(self):
+        """
+        Simply wait a bit and expect the registered request to be received
+        (and the corresponding response to be sent to the client).
+        """
+        logging.info(
+            f"--------------- RUN: {self.test_suite_id}, {self.test_case_id} ---------------"
+        )
+        time.sleep(5)
+        # access the previously registered callback
+        extended_diag_session_callback = uds_server_aux.callbacks["0x1003"]
```

Comment:
We are using the odx file in this case but I have to put raw byte like string to use it.

Is it possible to map what is inside the odx and have somthign a little bit more user friendly?

Reply:
Added registering of a second more readable key for odx based callbacks consisting of servicename.parameter (like "ReadDataByIdentifier.SoftwareVersion"). When unregistering one of those keys (hex version or readable version), I'm also deleting the other ones referencing the same callback, but I don't know if that is desirable.